resolwe/flow/views/collection.py

```diff
@@ -102,6 +106,69 @@ def get_queryset(self):
         """Prefetch entity data and return queryset."""
         return super().get_queryset().prefetch_related("entity_set")
 
+    @action(detail=True, methods=["post"])
+    def add_fields_to_collection(self, request, pk=None):
+        """Add the given list of AnnotaitonFields to the given collection."""
+        collection = self.get_object()
+        if not collection.has_permission(Permission.EDIT, request.user):
+            raise RuntimeError(
+                f"Must have edit permission on collection '{collection.slug}'."
+            )
+        serializer = AnnotationFieldsSerializer(data=request.data)
+        serializer.is_valid(raise_exception=True)
+        collection.annotation_fields.add(
+            *serializer.validated_data["annotation_fields"]
+        )
+        return Response()
+
+    @action(detail=True, methods=["post"])
+    def remove_fields_from_collection(self, request, pk=None):
+        """Remove the given list of AnnotaitonFields from the given collection.
+
+        The required argument is annotation_fields and optional confirm_action.
+        """
+        collection = self.get_object()
+        if not collection.has_permission(Permission.EDIT, request.user):
+            raise RuntimeError(
+                f"Must have edit permission on collection '{collection.slug}'."
+            )
+        serializer = AnnotationFieldsSerializer(data=request.data)
+        serializer.is_valid(raise_exception=True)
+
+        if not serializer.validated_data["confirm_action"]:
+            raise exceptions.ValidationError(
+                "All annotations for the given fields will be removed from the "
+                "samples in this collection. Please confirm."
+            )
+
+        annotation_fields = serializer.validated_data["annotation_fields"]
+        AnnotationValue.objects.filter(
+            entity__collection=collection, field__in=annotation_fields
+        ).delete()
+        collection.annotation_fields.remove(*annotation_fields)
+        return Response()
+
+    @action(detail=True, methods=["post"])
+    @transaction.atomic
+    def set_fields_on_collection(self, request, pk=None):
```

Comment:
Do we need to add a special method for setting fields, or is a combination of adding and removing sufficient?

Reply:
I would keep all of them to see which one will be easier to use for the frontend.

Reply:
After phone call I have removed the set method.